packages/react-charting/src/components/HorizontalBarChartWithAxis/HorizontalBarChartWithAxis.base.tsx

```diff
@@ -0,0 +1,614 @@
+        .padding(this.props.yAxisPadding || 0.1);
+
+      const xBarScale = d3ScaleLinear()
+        .domain(this._isRtl ? [xMax, 0] : [0, xMax])
+        .range([this.margins.left!, containerWidth - this.margins.right!]);
+      return { xBarScale, yBarScale };
+    }
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _createNumericBars(
+    containerHeight: number,
+    containerWidth: number,
+    xElement: SVGElement,
+    yElement: SVGElement,
+  ): JSX.Element[] {
+    const { useSingleColor = false } = this.props;
+    const { xBarScale, yBarScale } = this._getScales(containerHeight, containerWidth, true);
+    const colorScale = this._createColors();
+    const bars = this._points.map((point: IHorizontalBarChartWithAxisDataPoint, index: number) => {
+      let shouldHighlight = true;
+      if (this.state.isLegendHovered || this.state.isLegendSelected) {
+        shouldHighlight = this.state.selectedLegendTitle === point.legend;
+      }
+      this._classNames = getClassNames(this.props.styles!, {
+        theme: this.props.theme!,
+        legendColor: this.state.color,
+        shouldHighlight: shouldHighlight,
+      });
+      const barHeight: number = Math.max(yBarScale(point.y), 0);
+      if (barHeight < 1) {
+        return <React.Fragment key={point.x}> </React.Fragment>;
+      }
+      return (
+        <rect
+          key={point.y}
+          x={this._isRtl ? xBarScale(point.x) : this.margins.left!}
+          className={this._classNames.opacityChangeOnHover}
+          y={yBarScale(point.y) - this._barWidth / 2}
+          data-is-focusable={!this.props.hideTooltip}
+          width={
+            this._isRtl
+              ? containerWidth - this.margins.right! - Math.max(xBarScale(point.x), 0)
+              : Math.max(xBarScale(point.x), 0) - this.margins.left!
+          }
+          height={this._barWidth}
+          ref={(e: SVGRectElement) => {
+            this._refCallback(e, point.legend!);
+          }}
+          onClick={point.onClick}
+          onMouseOver={this._onBarHover.bind(this, point, colorScale(point.x))}
+          aria-label={this._getAriaLabel(point)}
+          role="img"
+          aria-labelledby={`toolTip${this._calloutId}`}
+          onMouseLeave={this._onBarLeave}
+          onFocus={this._onBarFocus.bind(this, point, index, colorScale(point.x))}
+          onBlur={this._onBarLeave}
+          fill={point.color && !useSingleColor ? point.color : colorScale(point.x)}
+        />
+      );
+    });
+    return bars;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _tooltipOfYAxislabels(ytooltipProps: any) {
+    const { tooltipCls, yAxis, id } = ytooltipProps;
+    if (yAxis === null) {
+      return null;
+    }
+    const div = d3Select('body').append('div').attr('id', id).attr('class', tooltipCls).style('opacity', 0);
+    const aa = yAxis!.selectAll('#BaseSpan')._groups[0];
```

Comment:
do we need to check the length of groups before calling 0.

Reply:
No , since we are already creating these groups in cartesian chart(label creation logic tooltip, wrap etc). So we dont need this check.